mkt/extensions/tests/test_validation.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+import json
+
+import mock
+from nose.tools import eq_
+from rest_framework.exceptions import ParseError
+from zipfile import ZipFile
+
+from django.core.files.uploadedfile import TemporaryUploadedFile
+
+from mkt.site.tests import TestCase
+from mkt.extensions.validation import ExtensionValidator
+
+
+class TestExtensionValidator(TestCase):
+    """
+    Tests the ExtensionValidator class. The following methods are tested in
+    the TestExtensionViewSetPost test case instead, as part of an end-to-end
+    workflow:
+
+    * ExtensionValidator.validate_file
+    * ExtensionValidator.validate_json
+    """
+    def setUp(self):
+        self.extension = None
+        self.validator = ExtensionValidator()
+        super(TestExtensionValidator, self).setUp()
+
+    def tearDown(self):
+        if self.extension:
+            self.extension.close()
+
+    def _extension(self, data):
+        self.extension = TemporaryUploadedFile('ext.zip', 'application/zip', 0,
+                                               'UTF-8')
+        with ZipFile(self.extension, "w") as z:
+            z.writestr('manifest.json', json.dumps(data))
+        return self.extension
+
+    def test_full(self):
+        extension = self._extension({
+            'name': 'My Extension',
+            'description': 'This is a valid description'
+        })
+        try:
+            ExtensionValidator(extension).validate()
+        except ParseError:
+            assert False, u'Valid extension failed validation.'
+
+    @mock.patch('mkt.extensions.validation.ExtensionValidator.validate_description')
+    @mock.patch('mkt.extensions.validation.ExtensionValidator.validate_name')
+    @mock.patch('mkt.extensions.validation.ExtensionValidator.validate_json')
+    @mock.patch('mkt.extensions.validation.ExtensionValidator.validate_file')
```

Comment:
`@mock.patch.object` helps, but any time you add a new validation method, you have to add it in 3 places, which kind of sucks. Not much to be done about it, I suppose.